File: 2026-01~02/JAVA/Data_Structure/stack/stack.java
class stack{

	int s[] = new int[100];
	int top = -1;
	final int max = 100;
	void error(final String msg){
		System.out.printf("[Error]: %s", msg);
	}

	public void push(int value){
		if(top == max - 1){
			//error("Stack is Full");
			return;
		}

		s[++top] = value;
	}

	public int pop(){
		if(top == -1){
	//		error("stack is Empty");
			return -1;
		}

		return s[top--];
	}
}

File: 24/Data Structure/Mid Exam/stack/stack.c
#include "stack.h"

void print_stack(STACK* s)
{
    for(int i = 0; i < MAX/2; i++)
    {
        if(i > s->top)
            printf("%3c|", ' ');
        else
            printf("%3d|",s->data[i]);

        if((i + 1) % 10 == 0)
            printf("\n");
    }
    printf("\n");

}

int main(void)
{
    STACK s;
    init_stack(&s);

    for(int i = 0; i < MAX/2; i++)
    {
        push(&s, i);
        print_stack(&s);
    }
    for(int i = 0; i < MAX/2; i++)
    {
        pop(&s);
        print_stack(&s);
    }
        
    return 0;
}
